mirrord-config/src/lib.rs

```diff
@@ -0,0 +1,228 @@
+#![feature(slice_concat_trait)]
+#![feature(once_cell)]
+
+pub mod agent;
+pub mod config;
+pub mod env;
+pub mod feature;
+pub mod fs;
+pub mod incoming;
+pub mod network;
+pub mod outgoing;
+pub mod pod;
+pub mod util;
+
+use std::path::Path;
+
+use mirrord_config_derive::MirrordConfig;
+use serde::Deserialize;
+
+use crate::{
+    agent::AgentField, config::source::MirrordConfigSource, feature::FeatureField, pod::PodField,
+    util::VecOrSingle,
+};
+
+#[derive(MirrordConfig, Deserialize, Default, PartialEq, Eq, Clone, Debug)]
+#[serde(deny_unknown_fields)]
+#[config(map_to = LayerConfig)]
+pub struct LayerFileConfig {
```

Comment:
This makes me think a new approach is probably needed, instead of making the result struct from the file config one to generate the file config struct from the result struct